documentation/cypress_cucumber.md

```diff
@@ -0,0 +1,133 @@
+step definitions in that file will be usable, e.g.:
+
+- `When I click on "change school link"`
+- `When I type "Bob" into field labelled "First name"`
+- `Then "page heading" should have value "Teacher: Bob"`
+
+#### Database
+
+We use [cypress-cucumber-preprocessor] to let us talk to factory_bot from our
+specs to set up simple state. E.g.:
+
+- `Given cohort was created with start_year "2021"`
+- `Given user was created as "admin" with email "admin@example.com"` (where
+  `admin` is the name of a trait).
+
+See the "Initialising database state" section below for more information on
+when this should and shouldn't be used.
+
+### Conventions
+
+#### Step definitions
+
+Nearly all of our existing step definitions that can be from the point of view
+of the user - e.g. what they do or what they should see - are in present tense,
+using active voice, and first-person.
+
+E.g.:
+
+- `Given I am on "index" page`
+- `When I click on "navigation link" containing "Dashboard"`
+- `Then I should be on "dashboard" page"`
+
+Exceptions are for steps that aren't from the point of the user, such as:
+
+- `Given scenario "user_cip" has been run`
+- `Then the page should be accessible`
+- `Then percy should be sent snapshot`
+
+#### Reuse step definitions where possible
+
+Most of the time, you won't need to write new step definitions as the existing
+ones are designed to be reusable by passing arguments into the steps: e.g.
+instead of `I click on user delete button` you can write `I click on "user
+delete" button` or `I click on "button" containing "delete user"` after adding
+the element to the objects in common-interaction.js.
+
+You should also consider this when writing new step definitions - make them as
+reusable as possible so that other people don't have to write similar step
+definitions in the future.
+
+#### Best practices
+
+[Cypress best practices] and Cucumber best practices sometimes conflict - e.g.
+Cucumber specs are supposed to be as small as possible and only test one thing
+while Cypress best practice is to test an entire user journey in one test as
+spinning up a browser and initialising the database can have a fairly
+significant overhead. It's good to be aware of both of them!
+
+#### Initialising database state
+
+For setting up simple data, the step definitions in `database.js` usually
+suffice, e.g.:
+
+- `Given cohort was created with start_year "2021"`
+
+For more complicated data, including pretty much anything relational, you'll
+want to use a scenario file instead. Create your scenario file (e.g. 
+`test_scenario.rb`) in `spec/cypress/app_commands/scenarios/` and use the
+following in your spec:
+
+- `Given scenario "test_scenario" has been run`
+
+#### Visual and accessibility testing
+
+We use [Percy] for visual testing and [cypress-axe] for basic automated
+accessibility testing. Every time you add a test for a new page (or significant
+state change to an existing page), add the following lines to your spec:
+
```

Comment:
Worth saying how to find out what the accessibility issue is when it fails? As in, Open the console and click on the step that was run before (on?) the check?